akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamSourceStage.scala

```diff
@@ -6,23 +6,23 @@ package akka.stream.impl.io
 
 import java.io.{ IOException, OutputStream }
 import java.util.concurrent.atomic.AtomicReference
-import java.util.concurrent.{ BlockingQueue, LinkedBlockingQueue }
+import java.util.concurrent.{ Semaphore, TimeUnit }
 
+import akka.dispatch.ExecutionContexts
 import akka.stream.Attributes.InputBuffer
 import akka.stream.impl.Stages.DefaultAttributes
 import akka.stream.impl.io.OutputStreamSourceStage._
 import akka.stream.stage._
-import akka.stream.{ ActorMaterializerHelper, Attributes, Outlet, SourceShape }
+import akka.stream.{ Attributes, Outlet, SourceShape }
 import akka.util.ByteString
 
+import scala.concurrent.Await
 import scala.concurrent.duration.FiniteDuration
-import scala.concurrent.{ Await, ExecutionContext, Future, Promise }
 import scala.util.control.NonFatal
-import scala.util.{ Failure, Success, Try }
 
 private[stream] object OutputStreamSourceStage {
   sealed trait AdapterToStageMessage
-  case object Flush extends AdapterToStageMessage
+  case class Send(data: ByteString) extends AdapterToStageMessage
   case object Close extends AdapterToStageMessage
 
   sealed trait DownstreamStatus
```

Comment:
Is DownstreamStatus still used?

Reply:
Yes, see below in `val downstreamStatus = new AtomicReference[DownstreamStatus](Ok)`